packages/sd-jwt/src/action-handler.ts

```diff
@@ -225,10 +225,29 @@ export class SDJwtPlugin implements IAgentPlugin {
     if (!payload.cnf) {
       throw Error('other method than cnf is not supported yet')
     }
-    const key = payload.cnf.jwk as JsonWebKey
-    return this.verifySignatureCallback(context)(data, signature, key)
+    return this.verifySignatureCallback(context)(data, signature, this.getJwk(payload))
   }
 
+  private getJwk(payload: JwtPayload): JsonWebKey {
+    if (payload.cnf?.jwk !== undefined) {
+      return payload.cnf.jwk as JsonWebKey
+    } else if (payload.cnf !== undefined && 'kid' in payload.cnf && typeof payload.cnf.kid === 'string' && payload.cnf.kid.startsWith('did:jwk:')) {
+      // extract JWK from kid FIXME isn't there a did function for this already? Otherwise create one
+      const encoded = this.extractBase64FromDID(payload.cnf.kid)
+      const decoded = decodeBase64url(encoded)
+      const jwt = JSON.parse(decoded)
+      return jwt as JsonWebKey
+    }
+    throw Error('Unable to extract JWK from SD-JWT payload')
+  }
+
+  private extractBase64FromDID(did: string): string {
+    const parts = did.split(':')
+    if (parts.length < 3) {
+      throw new Error('Invalid DID format')
```

Comment:
No this whole method should really not be used. This has nothing to do with DIDs in general. This is a did:jwk assumption, for which this handling is already odd to begin with

Reply:
It helped getting the sd-jwt verified for now. We do not supply cnf.jwk and without out the sd-jwt verification fails